lib/Sema/CSSyntacticElement.cpp

```diff
@@ -631,7 +631,7 @@ class SyntacticElementConstraintGenerator
   void visitForEachPattern(Pattern *pattern, ForEachStmt *forEachStmt) {
     auto target = SyntacticElementTarget::forForEachStmt(
         forEachStmt, context.getAsDeclContext(),
-        /*bindTypeVarsOneWay=*/false);
+        /*ignoreWhereClause=*/true);
```

Comment:
Would it be worth adding a comment here to explain that `where` clauses have already been handled by `visitForEachStmt`?

Reply:
Sure, I can add an explanation that points out that `visitForEachStmt` schedules a `where` clause as a separate element.

Reply:
Done!